simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsFileLargeDataSets.java

```diff
@@ -78,22 +81,24 @@ public BlockItem getNextBlockItem() throws IOException, BlockSimulatorParsingExc
     public Block getNextBlock() throws IOException, BlockSimulatorParsingException {
         currentBlockIndex++;
 
-        String nextBlockFileName = String.format(formatString, currentBlockIndex);
-        File blockFile = new File(blockstreamPath, nextBlockFileName);
-
-        if (!blockFile.exists()) {
+        final String nextBlockFileName = String.format(formatString, currentBlockIndex);
+        final Path localBlockStreamPath = Path.of(blockstreamPath).resolve(nextBlockFileName);
+        if (!Files.exists(localBlockStreamPath)) {
             return null;
         }
-
         try {
-            byte[] blockBytes = readFileBytes(blockFile.toPath());
+            final byte[] blockBytes =
+                    FileUtilities.readFileBytesUnsafe(
+                            localBlockStreamPath, RECORD_EXTENSION, GZ_EXTENSION);
 
-            LOGGER.log(INFO, "Loading block: " + blockFile.getName());
+            LOGGER.log(INFO, "Loading block: " + localBlockStreamPath.getFileName());
 
-            Block block = Block.PROTOBUF.parse(Bytes.wrap(blockBytes));
+            // todo blockBytes could be null, should we hande in some way or we need this method to
+            // fail here?
```

Comment:
TODO, does anyone have any opinion on this?

@jsync-swirlds @mattp-swirldslabs @georgi-l95

Reply:
In the _simulator_, I would recommend failing quickly.  The simulator should not attempt to continue if input could not be read.
We would need to log clearly what happened, of course.

Reply:
Yeah, maybe add a null check and a fail the simulator quickly with a message.

Reply:
Noted, I was thinking the same, I will throw here if null with a message (in the manner of "unable to read block file, most likely not found with given extensions", since we expect null in the else clause of the method, if the file is empty we should be getting a non-null empty array, the other option is an exception, which will happen before my check here, so is not within this scope) to the exception, hope that should be sufficient.

Reply:
✅ Done.